### PR TITLE
doc: add diagnosis memory leak guide

### PR DESCRIPTION
**Pull request by tony-go**
Hi Node Familly 👋 

## Context

The Diagnostic Working Group started an [initiative](https://github.com/nodejs/diagnostics/issues/532) to re-assess the [diagnostics tools list](https://github.com/nodejs/node/blob/master/doc/contributing/diagnostic-tooling-support-tiers.md).

I start to work around the `--trace-gc` flag:
* add it to the list itself https://github.com/nodejs/node/pull/42346
* add tests https://github.com/nodejs/node/pull/42471

But something is missing... 😞 

If you take a look at the diagnostic tooling support tiers, you'll see that:
> The tool must have a guide or other documentation in the Node.js GitHub organization or website;

## Added

I propose to add a new guide to the existing collection 🆕. 

The guide is about how could we use built-in tools to diagnose memory leaks.

Here is a list of topics I hope this guide will help on:
* basics of garbage collection
* what is the purpose of `--trace-gc`? and how could we interpret the output?
* how to perform heap snapshot?
* how to compare heap snapshot with Chrome dev tool?

**⚠️ DISCLAIMER**: I voluntarily simplify garbage collection concepts (that I didn't master myself). I don't know If it's a good idea or not, please share your thoughts on that. Note that my initial intention was to provide an introductory guide more than a deep dive.

## Concern

* I'm not sure about how should we handle images and gifs (location, size ...)

### CODE REVIEW

**Comment by tony-go** [comment]
cc @nodejs/diagnostics

File: locale/en/docs/guides/diagnosis-memory-leak.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+---
+# Diagnosis Memory Leak
+
+In this guide, you'll go through the fundamentals of memory leak diagnosis. And this, using built-in tools.
+
+## Setup
+
+For the proposal of this guide, we will use a simple web server empowered with [Fastify](https://www.fastify.io/).
+
+1 - Create a new project
```
Comment (nschonni) [inline locale/en/docs/guides/diagnosis-memory-leak.md]: Looks like this might be an ordered list

**Review by RafaelGSS — CHANGES_REQUESTED** [review]
I think the initial idea of the diagnostics working group was to add those guides in https://github.com/nodejs/diagnostics to then elaborate a long guide to nodejs.org.

I feel we need to discuss it. What do you think?

I recommend you add it to the next diag-agenda. cc: @nodejs/diagnostics 

EDIT1: I mean, I feel this guide would fit perfectly in: https://github.com/nodejs/diagnostics/tree/main/documentation/memory

**Comment by tony-go** [comment]
Hey @RafaelGSS 👋 

Sorry, I was off the past week so I didn't answer quickly.

> I think the initial idea of the diagnostics working group was to add those guides in https://github.com/nodejs/diagnostics to then elaborate a long guide to nodejs.org.
> 
> I feel we need to discuss it. What do you think?
> 
> I recommend you add it to the next diag-agenda. cc: @nodejs/diagnostics

[This issue](https://github.com/nodejs/diagnostics/issues/554) is still in the agenda we could discuss it at the next meeting 🙌  My only concern about keeping stuff in diag repo is that visibility/accessibility is limited. I'd like to expose each piece of the puzzle as soon as it's ready.

File: locale/en/docs/guides/diagnosis-memory-leak.md (hunk 1 above)
Comment (tony-go) [reply locale/en/docs/guides/diagnosis-memory-leak.md]: 🤦🏻‍♂️ Shame on me! I'll fix this ^^ nice catch.

**Review by Szolcsy — APPROVED** [review]

**Comment by tony-go** [comment]
As discussed in the last diagnostics WG meeting, we'll close this issue and merge the content into the diagnostic repository.

**Pull request closed without merging**